### PR TITLE
rt-app: Add support for SCHED_IDLE policy

### PR DESCRIPTION
**Pull request by vireshk**
This adds support for SCHED_IDLE scheduling policy.

This locally defines SCHED_IDLE macro, which is incorrect but build was
failing without it and I am not sure how to fix that. The arch specific
sched.h headers (like /usr/include/aarch64-linux-gnu/bits/sched.h)
define them only if __USE_GNU is defined.

Signed-off-by: Viresh Kumar <viresh.kumar@linaro.org>

### CODE REVIEW

File: src/rt-app_types.h [hunk 1]
```diff
@@ -42,9 +42,13 @@ Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 #define EXIT_INV_CONFIG 2
 #define EXIT_INV_COMMANDLINE 3
 
+/* It is not available in headers without defining __USE_GNU macro */
+#define SCHED_IDLE 5
```
Comment (vingu-linaro) [inline src/rt-app_types.h]: can you add a #ifndef __USE_GNU to prevent multiple definition ?

Comment (vireshk) [reply src/rt-app_types.h]: Okay,.

Comment (vireshk) [reply src/rt-app_types.h]: Vincent, I have made the change you were looking for. Thanks.

**Comment by vingu-linaro** [comment]
looks good to me

File: src/rt-app_types.h (hunk 1 above)
Comment (jlelli) [reply src/rt-app_types.h]: Could you rephrase to something like "SCHED_IDLE is not available if __USE_GNU is not defined" ?
Also, I'd update changelog, since we now have a reasonable solution to this problem. :-)

**Comment by vireshk** [comment]
Any updates on this @vingu-linaro  @jlelli  ?

**Comment by jlelli** [comment]
Thanks Viresh!
